cli/template/extras/src/server/auth/base.ts

```diff
@@ -37,6 +37,9 @@ export const authOptions: NextAuthOptions = {
   callbacks: {
     session({ session, user }) {
       if (session.user) {
+        // Check the shape of the data returned by the providers you use
+        // before adding other properties on the session.
+        // For example Google OAuth provider does not return an id for the user.
```

Comment:
are you sure about this?

pretty sure i've gotten an id back when using google oauth?

are you using jwt perhaps?

Reply:
I am also getting this error on a fresh project setup with the Discord Provider. The project can't build. 
```
Property 'id' does not exist on type '{ name?: string | null | undefined; email?: string | null | undefined; image?: string | null | undefined; }'.
```
Had to do a check for the key as well : `session.user && "id" in session.user`

Reply:
Yeah, like @Dodobrat I had to manually check the existence of the `id` field on the `user` object. Don't know why

Reply:
> I am also getting this error on a fresh project setup with the Discord Provider. The project can't build.
> 
> ```
> Property 'id' does not exist on type '{ name?: string | null | undefined; email?: string | null | undefined; image?: string | null | undefined; }'.
> ```
> 
> Had to do a check for the key as well : `session.user && "id" in session.user`
> 
> EDIT: Got it working by creating a new object, by spreading the existing session:
> 
> ```
> session({ session, user }) {
>   return {
>     ...session,
>     user: {
>       ...session.user,
>       id: user.id,
>     },
>   };
> }
> ```

I can't reproduce this. A new t3 app @ latest with all packages and the default discord auth setup builds fine

could you upload a minimal reproduction? ideally with the scaffolded app as the first commit, and any changes you make as another commit

Reply:
This was an issue with the latest version of next-auth, see issues page for more details. Downgrading next-auth to 4.20.0 helped me with the issue.

Reply:
That can't be the same issue OG faced though, the update came out yesterday and he had issues before that

Reply:
Hey @c-ehrlich , maybe the cli was updated or something but now I don't get the type error. 
-- Also forgot to mention that I got it after bumping up the dependencies. --

**Left**: New t3@latest project
**Right**: My project
 
![Screenshot 2023-04-03 at 11 18 13](https://user-images.githubusercontent.com/22771795/229453344-edf31efb-bb15-4f09-abab-6978a2d90654.jpg)

These are the the more important differences in my opinion. 
No point of sharing a new repo if it actually works. The error looks to be from `next-auth`, maybe not the same as @mikahFSmusic mentioned but the version I am on (Right side) is the issue

Reply:
> Hey @c-ehrlich , maybe the cli was updated or something but now I don't get the type error as of this morning. -- Also forgot to mention that I got it after bumping up the dependencies. --
> 
> **Left**: New t3@latest project **Right**: My project
> 
> ![Screenshot 2023-04-03 at 11 18 13](https://user-images.githubusercontent.com/22771795/229453344-edf31efb-bb15-4f09-abab-6978a2d90654.jpg)
> 
> These are the the more important differences in my opinion. No point of sharing a new repo if it actually works. The error looks to be from `next-auth`, maybe not the same as @mikahFSmusic mentioned but the version I am on (Right side) is the issue

Your problem was due to breaking changes in next-auth 4.21.0. You can see in the new app that we pinned it to 4.20.1 temporarily.